screens/EpisodeInputScreen.tsx

```diff
@@ -0,0 +1,398 @@
+    marginHorizontal: 15,
+    flexDirection: "row",
+    alignItems: "center",
+  },
+  icon: {
+    marginLeft: "auto",
+  },
+  text: {
+    fontFamily: "Arimo",
+    fontStyle: "normal",
+    fontWeight: "normal",
+    fontSize: 18,
+    lineHeight: 22,
+  },
+});
+
+export default function EpisodeInputScreen() {
+  let navigation = useNavigation();
+
+  const [episodeName, setEpisodeName] = useState("");
+  const [initials, setInitials] = useState("");
+  const [startTime, setStartTime] = useState<Date | undefined>(undefined);
+  const [endTime, setEndTime] = useState<Date | undefined>(undefined);
+
+  const [isStartDatePickerVisible, setStartDatePickerVisible] = useState(false);
+  const [isEndDatePickerVisible, setEndDatePickerVisible] = useState(false);
+
+  const [episodes, setEpisodes] = useState<Episode[]>([]);
+
+  let recordingDay: number = 1;
+
+  const addEpisodeToStoage = async (newEpisode: Episode) => {
+    const episodes = await AsyncStorage.getItem(
+      STORAGE_KEYS.daysEpisodes(recordingDay)
+    );
+    if (episodes) {
+      const episodeArray = JSON.parse(episodes);
+      episodeArray.push(newEpisode);
+      await AsyncStorage.setItem(
+        STORAGE_KEYS.daysEpisodes(recordingDay),
+        JSON.stringify(episodeArray)
+      );
+    }
+  };
+  const putAllEpisodesInStoage = async () => {
+    await AsyncStorage.setItem(
+      STORAGE_KEYS.daysEpisodes(recordingDay),
+      JSON.stringify(episodes)
+    );
+  };
+
+  const createEpisode = () => {
+    if (validateEpisode()) {
+      const newEpisode: Episode = {
+        name: episodeName,
+        initials,
+        startTime: convertToMinutes(startTime!),
+        endTime: convertToMinutes(endTime!),
+        date: CURRENT_DATE,
+        recordingDay,
+      };
+      episodes.push(newEpisode);
+      addEpisodeToStoage(newEpisode);
+      setEpisodes(episodes);
+      resetEpisodeInput();
+    }
+  };
+
+  const resetEpisodeInput = () => {
+    setEpisodeName("");
+    setInitials("");
+    setStartTime(undefined);
+    setEndTime(undefined);
+  };
+  const validateEpisode = () => {
+    return (
+      episodeName !== "" && startTime !== undefined && endTime !== undefined
+    );
+  };
+
+  const hasError = !validateEpisode();
+  const onMount = async () => {
+    const startDay = await AsyncStorage.getItem(STORAGE_KEYS.start_day());
+    const todayEpisodes = await AsyncStorage.getItem(
+      STORAGE_KEYS.daysEpisodes(recordingDay)
+    );
+    if (startDay === null) {
+      await AsyncStorage.setItem(STORAGE_KEYS.start_day(), CURRENT_DATE);
+    } else {
+      const msApart =
+        new Date(CURRENT_DATE).getMilliseconds() -
+        new Date(startDay).getMilliseconds();
+      recordingDay = msApart / DAY_IN_MS + 1;
+    }
+    if (todayEpisodes === null) {
+      await AsyncStorage.setItem(
+        STORAGE_KEYS.daysEpisodes(recordingDay),
+        JSON.stringify([])
+      );
+    } else {
+      setEpisodes(JSON.parse(todayEpisodes));
+    }
```

Comment:
Imma be real, I do not know what exactly is going on here, do you mind commenting in the code like what each of these things represent?